src/catalog/postgres/pg_core_impl.cpp

```diff
@@ -1525,6 +1558,119 @@ Column PgCoreImpl::MakeColumn(storage::ProjectedRow *const pr, const storage::Pr
   return col;
 }
 
+template <typename Column, typename ClassOid, typename ColOid>
+void PgCoreImpl::CreateColumnStatistic(const common::ManagedPointer<transaction::TransactionContext> txn,
+                                            const ClassOid class_oid, const ColOid col_oid, const Column &col) {
+  // Step 1: Insert into the table
+  auto *const redo = txn->StageWrite(db_oid_, PgStatistic::STATISTIC_TABLE_OID, pg_statistic_all_cols_pri_);
+
+  // Write the attributes in the Redo Record
+  auto relid_entry = reinterpret_cast<ClassOid *>(
```

Comment:
Could you look at how the new code handles accesses of projected rows?

Reply:
Done

Reply:
nice work. thanks!